technical_docs/Claims.md

```diff
@@ -0,0 +1,82 @@
+
+In this document, we outline the claims language utilized by SciTokens.  Inherent to the SciTokens concept is flexibility: any downstream user or VO may extend our claims with their own, domain-specific ones!
+
+Standard Claims
+---------------
+
+As a SciToken is a [JSON Web Token](https://jwt.io) at its base, we inherit a specific claims language from [RFC 7519](https://tools.ietf.org/html/rfc7519).  In this section, we outline the SciTokens-specific usage of the claims, denoting any changes in claim criticality.
+
+* *sub* (Subject): Typically indicates the individual or entity this token was originally issued to.  The subject is considered to be locally unique in the context of the VO.  Usage of this claim is OPTIONAL.  The token validators SHOULD NOT utilize the subject for authorization or identity mapping.  Suggested use cases are auditing, monitoring, or tracing.  Due to privacy concerns, some VOs may issue non-human-readable subjects or multiple anonymized subjects per individual.  A VO SHOULD NOT use the same subject for multiple individuals.
+
+* *nbf* (Not Before): The interpretation for `nbf` is unchanged from the RFC, but this is considered a CRITICAL attribute for a SciToken.
+
+* *exp* (Expiration Time): The interpretation for `exp` is unchanged from the RFC, but this is considered a CRITICAL attribute for a SciToken.  The motivation behind the combination of `nbf` and `exp` being marked as critical is to allow resources to enforce maximum token lifetime limits.
+
+* *iss* (Issuer): The issuer of the SciTokens.  VOs are strongly encouraged to populate the field; it MAY contain information local to the VO context (such as the host issuing the token).  SciTokens uses the `vo` claim name to identify the issuer in the global context.
+
+SciToken-specific Claims
+------------------------
+
+The SciTokens project defines JWT claims specific to our problem domain.  From RFC 7519:
+
+>   A producer and consumer of a JWT MAY agree to use Claim Names that
+>   are Private Names: names that are not Registered Claim Names
+>   (Section 4.1) or Public Claim Names (Section 4.2).  Unlike Public
+>   Claim Names, Private Claim Names are subject to collision and should
+>   be used with caution.
+
+In the long-term, we hope to have our claim names registered with IANA.  Until then, we will
+utilize the URI form.  A SciTokens validator MUST accept either URI or non-URI form.
+
+* *vo* (Virtual Organization): https://scitokens.org/v1/vo. The virtual organization issuing the token.  This MUST be globally unique for the community utilizing the SciToken.  It MUST be utilized to determine the appropriate trust roots for token verification.  If present in a token, it MUST also be present in the header.  A token chain MUST have the VO specified in the base token.
```

Comment:
I was thinking of having this be a field for internal information from the VO.  I agree it makes more sense to have it use my standard OAuth-style.  Updating...